### PR TITLE
Feature/ask command

### PR DESCRIPTION
**Pull request by Pete-Senpai**
Pull Request Summary

Merged branch 'feature/ask-command' into the main branch

Commits (in reverse order):

6. Added the /ask command
   - This commit adds the '/ask' command.

5. Updated tell command by Pete-Senpai on Sep 18, 2023
   - This commit updates the 'tell' command.

4. Added scanForKeyword by Pete-Senpai on Sep 19, 2023
   - This commit adds the 'scanForKeyword' feature.

3. Refactored interaction to message ref by Brakistad on Sep 19, 2023
   - This commit refactors the interaction to message reference.

2. Refactor namingfunc by Brakistad on Sep 20, 2023
   - This commit refactors the 'namingfunc'.

1. Enabled scanForKeyword to work by Pete-Senpai on Sep 20, 2023
   - This commit enables the functionality of 'scanForKeyword'.

Thank you for your contributions!

### CODE REVIEW

File: bot/bot.js [hunk 1]
```diff
@@ -224,36 +233,72 @@ client.on('interactionCreate', async interaction => {
...
+    } else {
+        logger.debug("AI is enabled");
     }
+    
+
+    // Message content
+    const content = message.content
+
+    // Message author
+    const author = message.author.global_name
```
Comment (Brakistad) [inline bot/bot.js:261]: This variable is optional, per docs, and should be handled.
```suggestion
    var author = message.author.global_name
    if (author == undefined) {
        author = message.author.username // as a fallback
    }
```
}

File: bot/bot-refresh.js [hunk 2]
```diff
@@ -22,6 +22,13 @@ const commands = [
             option.setName('mode')
                 .setDescription('The mode to use\nAvailable modes: default, sys_admin, juridisk_megler')
                 .setRequired(true)),
+    new SlashCommandBuilder()
+        .setName('tell')
+        .setDescription('Tell Chappie')
+        .addStringOption(option =>
+            option.setName('message')
+                .setDescription('tell Chappie something')
+                .setRequired(true)), // Define the /ask command with a required question option
```
Comment (Brakistad) [inline bot/bot-refresh.js:31]: Maybe remove? if not used

File: bot/openai-request.js [hunk 3]
```diff
@@ -13,6 +13,7 @@ if (require.main === module) {
 
 // Make a request to the OpenAI API
 function openaiRequest(model, bot_context, callback) {
+    logger.debug(bot_context);
```
Comment (Brakistad) [inline bot/openai-request.js:16]: This is very verbose, maybe fine on debug.

File: bot/bot.js (hunk 1 above)
Comment (Pete-Senpai) [reply bot/bot.js:261]: Okay, it has been changed

File: bot/bot-refresh.js (hunk 2 above)
Comment (Pete-Senpai) [reply bot/bot-refresh.js:31]: Also done

File: bot/openai-request.js (hunk 3 above)
Comment (Pete-Senpai) [reply bot/openai-request.js:16]: It has been deleted :)

File: bot/bot.js [hunk 4]
```diff
@@ -224,36 +233,74 @@ client.on('interactionCreate', async interaction => {
...
+        logger.debug("true");
+    }
 
-    // Add message to history if it is not from the bot
+    if (!ai_enabled) {
+        logger.info("AI is not enabled");
+        return;
+    } else {
+        logger.debug("AI is enabled");
+    }
```
Comment (Brakistad) [inline bot/bot.js:302]: This code is a duplicate from the one in top of method

Comment (Pete-Senpai) [reply bot/bot.js:302]: Complete

**Review by Brakistad — APPROVED** [review]

**Pull request merged**